File: examples/echo/echo.js
//----------------------------------------------------------------------------------------------------------------------
// A simple echo test.
//
// Example code licensed under WTFPL(http://www.wtfpl.net/).
//
// @module echo.js
//----------------------------------------------------------------------------------------------------------------------

var NodePy = require('nodepy');

//----------------------------------------------------------------------------------------------------------------------

function echo()
{
    setTimeout(function()
    {
        pythonWorker.emit('echo', "Test for echo...");
    }, 1000);
} // end echo

//----------------------------------------------------------------------------------------------------------------------

console.log("NodePy Echo Test starting...");

// Setup our python process
var pythonWorker = new NodePy('./echo.py');

// Listen for echo back from the python process
pythonWorker.on('echo', function(data)
{
    console.log("[node] Got:", data);

    // Schedule another echo
    echo();
});

// Start our python process
pythonWorker.start();

// Start sending echos
echo();

//----------------------------------------------------------------------------------------------------------------------

File: examples/echo/echo.py
#-----------------------------------------------------------------------------------------------------------------------
# A simply python echo worker.
#
# Example code licensed under WTFPL(http://www.wtfpl.net/).
#
#-----------------------------------------------------------------------------------------------------------------------

from pynode import Bridge

#-----------------------------------------------------------------------------------------------------------------------

class EchoWorker(Bridge):
    def __init__(self):
        super(EchoWorker, self).__init__()

        # Register for the echo event
        self.on('echo', self.handleEcho)

    def handleEcho(self, data):
        print "[python] Got:", repr(data)

        self.emit('echo', data)

# Create our worker
worker = EchoWorker()

# Loop, waiting for input
worker.loop()